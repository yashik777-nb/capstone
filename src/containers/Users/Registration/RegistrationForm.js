import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Registration from "../../../components/Registration/Registration";
import * as actionCreator from "../../../store/actions/actionsIndex";
class RegistrationForm extends React.Component {
  registerUserHandler(user) {
    this.props.registerUser(user);
    this.props.history.push("/");
  }
  render() {
    return <Registration onSave={(user) => this.registerUserHandler(user)} />;
  }
}

const mapDispathToProps = (dispatch) => {
  return {
    registerUser: (user) => dispatch(actionCreator.addUser(user)),
  };
};

export default connect(null, mapDispathToProps)(withRouter(RegistrationForm));
