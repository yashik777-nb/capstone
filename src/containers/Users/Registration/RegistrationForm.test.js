import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import RegistrationForm from "./RegistrationForm";

configure({ adapter: new Adapter() });

const props = {
  registerUser: () => {},
};

const store = {};
const mockStore = configureStore();

describe("Registration Form Container Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <RegistrationForm {...props} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should render Registration Form", () => {
    expect(wrapper.find("Registration").length).toEqual(1);
  });

  test("Should Render Registration Form Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
