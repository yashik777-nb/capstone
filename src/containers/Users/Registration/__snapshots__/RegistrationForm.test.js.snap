// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Container Tests Should Render Registration Form Snapshot 1`] = `ReactWrapper {}`;
